NeosModLoader/ModLoaderConfiguration.cs

```diff
@@ -1,73 +1,113 @@
+
+			return Path.GetDirectoryName(path);
+		}
+
+		private static string? GetConfigPath()
+		{
+			if (LaunchArguments.TryGetArgument("Config.Path", out var argument))
+				return argument.Value;
+
+			// the config file can just sit next to the dll. Simple.
+			return Path.Combine(GetAssemblyDirectory(), CONFIG_FILENAME);
+		}
 
+		private static ModLoaderConfiguration LoadConfig()
+		{
+			var path = GetConfigPath();
+			var config = new ModLoaderConfiguration();
+
+			var configOptions = typeof(ModLoaderConfiguration).GetProperties(AccessTools.all).ToArray();
+
+			if (!File.Exists(path))
+			{
+				Logger.MsgInternal($"Using default config - file doesn't exist: {path}");
+			}
+			else
+			{
 				// .NET's ConfigurationManager is some hot trash to the point where I'm just done with it.
 				// Time to reinvent the wheel. This parses simple key=value style properties from a text file.
 				try
 				{
 					var lines = File.ReadAllLines(path);
 					foreach (var line in lines)
 					{
-						int splitIdx = line.IndexOf('=');
-						if (splitIdx != -1)
+						var splitIdx = line.IndexOf('=');
+						if (splitIdx == -1)
+							continue;
+
+						string key = line.Substring(0, splitIdx).Trim();
+						string value = line.Substring(splitIdx + 1).Trim();
+
+						var possibleProperty = configOptions.FirstOrDefault(property => property.Name.Equals(key, StringComparison.InvariantCultureIgnoreCase));
+						if (possibleProperty == null)
 						{
-							string key = line.Substring(0, splitIdx);
-							string value = line.Substring(splitIdx + 1);
-
-							if ("unsafe".Equals(key) && "true".Equals(value))
-							{
-								_configuration.Unsafe = true;
-							}
-							else if ("debug".Equals(key) && "true".Equals(value))
-							{
-								_configuration.Debug = true;
-							}
-							else if ("hidevisuals".Equals(key) && "true".Equals(value))
-							{
-								_configuration.HideVisuals = true;
-							}
-							else if ("nomods".Equals(key) && "true".Equals(value))
-							{
-								_configuration.NoMods = true;
-							}
-							else if ("nolibraries".Equals(key) && "true".Equals(value))
-							{
-								_configuration.NoLibraries = true;
-							}
-							else if ("advertiseversion".Equals(key) && "true".Equals(value))
-							{
-								_configuration.AdvertiseVersion = true;
-							}
-							else if ("logconflicts".Equals(key) && "false".Equals(value))
-							{
-								_configuration.LogConflicts = false;
-							}
-							else if ("hidemodtypes".Equals(key) && "false".Equals(value))
-							{
-								_configuration.HideModTypes = false;
-							}
-							else if ("hidelatetypes".Equals(key) && "false".Equals(value))
-							{
-								_configuration.HideLateTypes = false;
-							}
+							Logger.WarnInternal($"Unknown key found in config file: {key}");
+							Logger.WarnInternal($"Supported keys: {string.Join(", ", configOptions.Select(property => property.Name))}");
+							continue;
```

Comment:
Seems like this would get repeated for each unknown key due to being in the loop, would rather just store the found unknown keys and log them all in one row after the loop.